numba/numpy_support.py

```diff
@@ -123,6 +123,8 @@ def as_dtype(nbtype):
     if isinstance(nbtype, (types.CharSeq, types.UnicodeCharSeq)):
         letter = _as_dtype_letters[type(nbtype)]
         return numpy.dtype('%s%d' % (letter, nbtype.count))
+    if isinstance(nbtype, types.Record):
```

Comment:
I don't think this is required - I think it would only be hit if you had a record within a record, which is not supported at the moment.